ethereum/eth/src/main/java/org/hyperledger/besu/ethereum/eth/manager/snap/SnapServer.java

```diff
@@ -456,8 +456,9 @@ MessageData constructGetBytecodesResponse(final MessageData message) {
         } else {
           Optional<Bytes> optCode = worldStateStorageCoordinator.getCode(Hash.wrap(codeHash), null);
           if (optCode.isPresent()) {
-            if (sumListBytes(codeBytes) + optCode.get().size() > maxResponseBytes
-                || stopWatch.getTime() > StatefulPredicate.MAX_MILLIS_PER_REQUEST) {
+            if (!codeBytes.isEmpty()
```

Comment:
sumListBytes operates on the stream, with an orElse(0).  it shouldn't matter whether the list is empty or not right?  what am I missing here?

Reply:
The sumListBytes will work fine on an empty list. This change isn't to guard against calling sumListBytes with an empty list but to ensure we only do the `sumListBytes(codeBytes) > maxReponseBytes` check once we have at least one record in the response. So that we always send at least once record in the response.